static-site/src/pages/index.jsx

```diff
@@ -10,6 +10,10 @@ import MainLayout from "../components/layout";
 
 class Index extends React.Component {
   render() {
+    if (config.groupsOnlyService) {
+      return "This should redirect to /groups"
+    }
```

Comment:
If we aren't going to solve the client-side routing issue for now, then we could recapitulate the redirect here with something like:

```suggestion
    if (config.groupsOnlyService) {
      navigate("/groups");
      return;
    }
```

where `navigate()` is from `import { navigate } from "gatsby"`. Though this doesn't immediately work—it changes the address bar but the page remains blank—for unclear and uninvestigated reasons. Possibly related to?

https://github.com/nextstrain/nextstrain.org/blob/9cd74b7fd7c2fb445f6a52679c1ffc3475fe5d8e/src/setupGroups.js#L51-L54

An alternative is to bypass Gatsby's `navigate()` (and thus client router) with:

```diff
 class Index extends React.Component {
   render() {
-    if (groupsOnlyService) {
-      return "This should redirect to /groups"
-    }
-
     // Workaround so index page doesn't flash when pages are redirecting
     // See github issue: https://github.com/gatsbyjs/gatsby/issues/5329#issuecomment-484741119
     const browser = typeof window !== "undefined" && window;
 
+    if (groupsOnlyService) {
+      if (browser) {
+        window.location = "/groups";
+      }
+      return;
+    }
+
```

And maybe another option is a `createRedirect()` call in `gatsby-node.js`.

Reply:
Thanks for the different options. I got Gatsby's `navigate()` to work ([preview](https://nextstrain-s-victorlin--6ooibb.herokuapp.com)):

https://github.com/nextstrain/nextstrain.org/blob/5db4236e644dc01d60538921d10230897c965681/static-site/src/pages/index.jsx#L17-L20